quarkus/runtime/src/main/java/org/keycloak/quarkus/runtime/cli/command/Start.java

```diff
@@ -56,6 +58,10 @@ protected void doBeforeRun() {
         if (Environment.isDevProfile()) {
             throw new PropertyException(Messages.devProfileNotAllowedError(NAME));
         }
+        if (Configuration.getConfigValue(DB).getConfigSourceOrdinal() == 0) {
+          picocli.warn("Usage of the default value for the db option in the production profile is deprecated. Please explicitly set the db instead.");
```

Comment:
Duplicates the build warning (seen with auto-build or build). Having it here is possible too, it implies that in some instances users are unaware of what the optimized images is configured for.